File: src/dere_plugin/skills/detecting-user-emotions/scripts/format_emotion.py
#!/usr/bin/env python3
"""Format emotion state for human-readable output."""

from __future__ import annotations

import json
import sys


def format_emotion(state: dict) -> str:
    """Format emotion state into readable text."""
    lines = []

    primary = state.get("primary", {})
    if primary:
        name = primary.get("name", "unknown")
        intensity = primary.get("intensity", 0)
        lines.append(f"Primary: {name} ({intensity}% intensity)")

    secondary = state.get("secondary", {})
    if secondary:
        name = secondary.get("name", "unknown")
        intensity = secondary.get("intensity", 0)
        lines.append(f"Secondary: {name} ({intensity}% intensity)")

    trust = state.get("trust_level", 0)
    lines.append(f"Trust: {trust:.0%}")

    return "\n".join(lines)


def main():
    """Main entry point."""
    try:
        state = json.load(sys.stdin)
        print(format_emotion(state))
    except json.JSONDecodeError:
        print("Error: Invalid JSON input", file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()

File: src/dere_plugins/dere_core/skills/mood/scripts/format_emotion.ts
function formatEmotion(state: Record<string, unknown>): string {
  const lines: string[] = [];

  const primary = state.primary as Record<string, unknown> | undefined;
  if (primary && typeof primary === "object") {
    const name = (primary.name as string | undefined) ?? "unknown";
    const intensity = (primary.intensity as number | undefined) ?? 0;
    lines.push(`Primary: ${name} (${intensity}% intensity)`);
  }

  const secondary = state.secondary as Record<string, unknown> | undefined;
  if (secondary && typeof secondary === "object") {
    const name = (secondary.name as string | undefined) ?? "unknown";
    const intensity = (secondary.intensity as number | undefined) ?? 0;
    lines.push(`Secondary: ${name} (${intensity}% intensity)`);
  }

  const trust = typeof state.trust_level === "number" ? state.trust_level : 0;
  lines.push(`Trust: ${(trust * 100).toFixed(0)}%`);

  return lines.join("\n");
}

async function main(): Promise<void> {
  try {
    const input = await Bun.stdin.text();
    const state = JSON.parse(input) as Record<string, unknown>;
    console.log(formatEmotion(state));
  } catch {
    console.error("Error: Invalid JSON input");
    process.exit(1);
  }
}

if (import.meta.main) {
  void main();
}
